Add analytics 1

Aichnerc opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently, there is no Google Analytics, Facebook Pixel or Hotjar implemented.


**What is the new behavior (if this is a feature change)?**
- Google Analytics, Facebook Pixel and Hotjar has been implemented.


**Other information**:
- The version has been updated to v1.5.1.

kleberbaum commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -111,8 +113,8 @@ class Footer extends React.Component{
                         <MDBCol md="2">
-                            <h5>Logo</h5>
-                            <p>
+                            <img src={Logo} style={{ height: "2.5rem" }} />
```
Is it absolutely necessary to define this style in html?

Aichnerc replied:
Nah, I just copied it from the MDB Pro documentation. But you're right, it's way cleaner to define it in footer.scss.

kleberbaum commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -157,20 +159,20 @@ class Footer extends React.Component{
                             </li>
                             <li className="list-unstyled">
-                                <a href="#!"><MDBIcon far icon="envelope" />info@aichner-christian.com</a>
+                                <a href="mailto:info@aichner-christian.com"><MDBIcon far icon="envelope" />info@aichner-christian.com</a>
```
Unnecessarily long line.
We allow up to 119 characters as this is the width of GitHub code review.

Aichnerc replied:
The entire file has been cleaned up regarding line length.

Aichnerc committed:
Implement hotjar

Hotjar tracking service has been added.

Aichnerc committed:
Implement basic Google Analytics

The basic Google Analytics has been added.

Aichnerc committed:
Fix Link exact

An issue with links has been fixed.

Aichnerc committed:
Improve Analytics

The analytics has been implemented in the ContactForm and CallToAction sections.

Aichnerc committed:
Improve Analytics

The analytics has been added to the Analysis page.

Aichnerc committed:
Fix Analysis crash

The Google Analytics functions have been added to the Analysis page.

Aichnerc committed:
Improve Analytics init

The init should now happen as intended.

Aichnerc committed:
Implement Facebook Pixel

The basic Facebook pixel has been implemented.

Aichnerc committed:
Add Facebook Pixel events

Some basic events have been added.

Aichnerc committed:
Update FB Pixel

The trackCustom function has been used.

Aichnerc committed:
Update version

The version has been updated.

Aichnerc committed:
Improve code quality

Some line issues have been fixed.

Aichnerc committed:
Improve code qualiy

Some indents have been fixed.

LucaAllmaier reviewed (approved):
Looks good to me.

Aichnerc committed:
Improve Google Analytics

Some category has been changed.

Aichnerc reviewed (commented):
Looking good to my old eyes.

sweggerat reviewed (approved):
looks good

pinterid commented on src/App.js:
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
+
+  checkCookies = () => {
+    // Create custom user id for tracking
+    let userId = localStorage.getItem("userId");
```
Add newline

pinterid commented on src/App.js:
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
+    }
+
+    // Check cookies
+    let cookie = localStorage.getItem("cookie");
```
Add newline.

pinterid commented on src/App.js:
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
+    // Check cookies
+    let cookie = localStorage.getItem("cookie");
+    if (cookie) {
+      cookie = JSON.parse(cookie);
```
Add newline.

pinterid commented on src/App.js:
```diff
@@ -41,6 +47,99 @@ class App extends React.Component {
+    let userId = localStorage.getItem("userId");
+    if (!userId) {
+      const sha256 = require("js-sha256");
+      userId = sha256.create();
```
Add newline

schettn commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -70,6 +84,9 @@ class ContactForm extends React.Component {
 
+    // Analytics
+    this.props.googleAnalytics.registerContactSend();
+
```
It this newline wanted? In my opinion there is no reason to separate those two function calls.

pinterid commented on src/components/organisms/sections/Hero/index.jsx:
```diff
@@ -160,7 +163,10 @@ class HomePage extends React.Component {
                   aktuellen Stand Deiner gesamten Online-Präsenz und liefert
```
I would add a "Dir" after "liefert".

schettn commented on src/config/fbInit.js:
```diff
@@ -9,10 +9,13 @@ import config from "./fbConfig.js";
+ * Initialize Google Analytics
+ * Once we implement it ourselves, this is not required for now
+ */
+/*if (config.measurementId) {
```
I would recommend to comment out this section using //

Aichnerc replied:
I just removed the entire thing. I won't be using it anymore.

schettn commented on src/config/fbInit.js:
```diff
@@ -9,10 +9,13 @@ import config from "./fbConfig.js";
 firebase.initializeApp(config);
-// Init analytics
-if (config.measurementId) {
+/**
```
Is it wanted that you use /** */ as header? Because every other header starts with //>.

pinterid reviewed (changes requested):
Looks good after some changes.

schettn reviewed (changes requested):
Please check.

Aichnerc committed:
Improve code quality

A few changes have been made by a request of @pinterid.

Aichnerc committed:
Improve code quality

Some unused code has been removed.

schettn reviewed (approved):
🚀

pinterid reviewed (approved):
Looks good to merge.
